### ISSUE TITLE
Wallet got stuck

### ISSUE THREAD

**Issue by alexandrz** [spesmilo/electrum#1479]
I am getting this error, and nothing happens after it.

```
[Synchronizer] receiving history 16QxzkfYoXV5fqE8R9Uvk7Vsy72yYN1C6u 0
removing tx from history 1dcf97aee12900575eeaf1443b8f78f934a38106d631177f2e9875223339c2c1
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/local/Cellar/python/2.7.8_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/local/lib/python2.7/site-packages/electrum/network_proxy.py", line 78, in run
    self.process(response)
  File "/usr/local/lib/python2.7/site-packages/electrum/network_proxy.py", line 128, in process
    callback(r)
  File "/usr/local/lib/python2.7/site-packages/electrum/synchronizer.py", line 112, in addr_history_response
    self.wallet.receive_history_callback(addr, hist)
  File "/usr/local/lib/python2.7/site-packages/electrum/wallet.py", line 787, in receive_history_callback
    self.remove_transaction(tx_hash)
  File "/usr/local/lib/python2.7/site-packages/electrum/wallet.py", line 770, in remove_transaction
    self.txi.pop(tx_hash)
KeyError: u'1dcf97aee12900575eeaf1443b8f78f934a38106d631177f2e9875223339c2c1'

[electrum.drollette.com] failed to connect [Errno 60] Operation timed out
[Network] network: retrying connections
[electrum3.hachre.de] connected
```

**Comment by ecdsa** [comment]
what version of Electrum is that?

**Comment by alexandrz** [comment]
2.4.4

**Comment by alexandrz** [comment]
Getting the same error in 2.5

```
[46.166.165.18] serving improperly sorted address histories
[wallet3] removing tx from history c9ffd7ed768198aea2a9f14aa77c454cf2c632d62a88d0f29cfc64fa5d349914
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/local/Cellar/python/2.7.8_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/local/lib/python2.7/site-packages/Electrum-2.5-py2.7.egg/electrum/network.py", line 723, in run
    self.wait_on_sockets()
  File "/usr/local/lib/python2.7/site-packages/Electrum-2.5-py2.7.egg/electrum/network.py", line 717, in wait_on_sockets
    self.process_responses(interface)
  File "/usr/local/lib/python2.7/site-packages/Electrum-2.5-py2.7.egg/electrum/network.py", line 531, in process_responses
    self.process_response(interface, response, callback)
  File "/usr/local/lib/python2.7/site-packages/Electrum-2.5-py2.7.egg/electrum/network.py", line 493, in process_response
    callback(response)
  File "/usr/local/lib/python2.7/site-packages/Electrum-2.5-py2.7.egg/electrum/synchronizer.py", line 110, in addr_history_response
    self.wallet.receive_history_callback(addr, hist)
  File "/usr/local/lib/python2.7/site-packages/Electrum-2.5-py2.7.egg/electrum/wallet.py", line 789, in receive_history_callback
    self.remove_transaction(tx_hash)
  File "/usr/local/lib/python2.7/site-packages/Electrum-2.5-py2.7.egg/electrum/wallet.py", line 772, in remove_transaction
    self.txi.pop(tx_hash)
KeyError: u'c9ffd7ed768198aea2a9f14aa77c454cf2c632d62a88d0f29cfc64fa5d349914'
```

**Comment by ecdsa** [comment]
in the python console, type: wallet.clear_history()
then exit ans restart Electrum
please report here about results

**Comment by alexandrz** [comment]
Yes, it resolved the issue. I think I did the same by removing and restoring the wallet before. It works for some time and then it gets stuck again with the same error (but another transaction).

This wallet has more than 3000 addresses and receives about 50 transactions per day.

**Comment by ecdsa** [comment]
as far as I understand, it got stuck with an old version.
please reopen this bug it it gets stuck again with 2.5.

**Comment by alexandrz** [comment]
It's happened again:

```
[wallet3] removing tx from history fde5c9c893eac7465f7f5f2551e9cef6e00f97f67c8804ca9b0b7bdbeacc1524
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/local/Cellar/python/2.7.8_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/local/lib/python2.7/site-packages/Electrum-2.5-py2.7.egg/electrum/network.py", line 723, in run
    self.wait_on_sockets()
  File "/usr/local/lib/python2.7/site-packages/Electrum-2.5-py2.7.egg/electrum/network.py", line 717, in wait_on_sockets
    self.process_responses(interface)
  File "/usr/local/lib/python2.7/site-packages/Electrum-2.5-py2.7.egg/electrum/network.py", line 531, in process_responses
    self.process_response(interface, response, callback)
  File "/usr/local/lib/python2.7/site-packages/Electrum-2.5-py2.7.egg/electrum/network.py", line 493, in process_response
    callback(response)
  File "/usr/local/lib/python2.7/site-packages/Electrum-2.5-py2.7.egg/electrum/synchronizer.py", line 110, in addr_history_response
    self.wallet.receive_history_callback(addr, hist)
  File "/usr/local/lib/python2.7/site-packages/Electrum-2.5-py2.7.egg/electrum/wallet.py", line 789, in receive_history_callback
    self.remove_transaction(tx_hash)
  File "/usr/local/lib/python2.7/site-packages/Electrum-2.5-py2.7.egg/electrum/wallet.py", line 772, in remove_transaction
    self.txi.pop(tx_hash)
KeyError: u'fde5c9c893eac7465f7f5f2551e9cef6e00f97f67c8804ca9b0b7bdbeacc1524'
```

**Comment by alexandrz** [comment]
Now I am getting this error:

```
[wallet3] removing tx from history 7b250c062d2625acb5dd1b4c52db6868911bd902d07b741ccab7deef116bae24
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/local/Cellar/python/2.7.8_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/local/lib/python2.7/site-packages/Electrum-2.5.1-py2.7.egg/electrum/network.py", line 723, in run
    self.wait_on_sockets()
  File "/usr/local/lib/python2.7/site-packages/Electrum-2.5.1-py2.7.egg/electrum/network.py", line 717, in wait_on_sockets
    self.process_responses(interface)
  File "/usr/local/lib/python2.7/site-packages/Electrum-2.5.1-py2.7.egg/electrum/network.py", line 531, in process_responses
    self.process_response(interface, response, callback)
  File "/usr/local/lib/python2.7/site-packages/Electrum-2.5.1-py2.7.egg/electrum/network.py", line 493, in process_response
    callback(response)
  File "/usr/local/lib/python2.7/site-packages/Electrum-2.5.1-py2.7.egg/electrum/synchronizer.py", line 110, in addr_history_response
    self.wallet.receive_history_callback(addr, hist)
  File "/usr/local/lib/python2.7/site-packages/Electrum-2.5.1-py2.7.egg/electrum/wallet.py", line 791, in receive_history_callback
    self.remove_transaction(tx_hash)
  File "/usr/local/lib/python2.7/site-packages/Electrum-2.5.1-py2.7.egg/electrum/wallet.py", line 775, in remove_transaction
    except KeyErrror:
NameError: global name 'KeyErrror' is not defined
```

**Comment by alexandrz** [comment]
Should be fixed by fc141cb
